### ISSUE TITLE
FreeBSD Support for 3.4.0

### ISSUE THREAD

**Issue by y2kbadbug** [Guake/guake#1444]
I have spent several days working on updating the FreeBSD port for Guake 3.4.0.
I am using the sdist in PyPi which in FreeBSD I believe is CHEESESHOP.

A few notes about the patches:
* I had to patch all instances of
`from locale import gettext as _`
to
`from gettext import gettext as _`

* I had to patch the utempter.so location.  It was searching for utempter.so.0 and the library is utempter.so.  There is no port for utempter as it's built into the system.

* I had to modify the paths.py file, which I took from a Gentoo patch (This fixed the schema issues)
https://github.com/gentoo/gentoo/blob/master/x11-terms/guake/files/guake-3.4.0-paths.patch

* I had to patch the Makefile to make  PYTHON_INTERPRETER=python3 => ${PYTHON_CMD}.
I did not compile from source so I did not have to worry about pipenv, however, pipenv wouldn't work as it needs specific version such as pipenv-${PYTHON_VERSION}.  As mentioned, I did not modify these as they are not required to compile.

Attached is a copy of the updated port.  I will be submitting it to review later tonight.  After discussing this with the FreeBSD Python team, it should be up to par.
[x11_guake.zip](https://github.com/Guake/guake/files/2466124/x11_guake.zip)

Taking the latest version current on FreeBSD is 0.8.10, this is a major upgrade.

I've also attached the poudriere build log for anyone interested.
[guake-3.4.0.log](https://github.com/Guake/guake/files/2466129/guake-3.4.0.log)

**Comment by gsemet** [comment]
Thanks for your feedback.

Why is that `from locale import gettext` does not work on freebsd? If the `gettext` package work the same, I do not see the reason not to use it by default in guake directly. 

For the utempter.so, I do not see any other way than add a "if freebsd" test in the code (I do not know the system so I let you tell me which is the best test for that, or do a pull request for that). Or maybe 2 tries, brutally. Might work.

For paths.py, I'll consider indeed your remark to ditch the special handling for the data files and load directly from the python package directly. Wonder how I'll manage the gsetting schema and the translation.

There is no problem in building without pipenv :)

**Comment by y2kbadbug** [comment]
I am not quite sure why `from locale import gettext as _` does not work.  I did some troubleshooting with it and never could actually see why.  It just didn't seem to find the gettext module through locale.  Possibly a change with Python 3.6?  Locale and gettext I believe are builtin modules

Patching the file for utempter.so using the port system might just be the best solution for right now since it's just a small patch and seems to be isolated to FreeBSD.  In all honesty, building software outside of the port systems is discouraged and unsupported.  This is all based on my own opinion and is open to suggestions.

I had no intentions of wanting to ditch the original method for paths.py, however, it just simply does not work on FreeBSD (and other systems it appears).  I tried to troubleshoot this for hours and could not get the gschemas to work correctly any other way.

I submitted my request to update the port to FreeBSD and can be tracked here:
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=232163

Hopefully, there isn't too much pushback on it.  I was able to run Guake on my laptop with Gnome 3.28 on FreeBSD 12-ALPHA8 and it ran real smooth.

Thanks for your feedback as well.

I think this issue can be closed once FreeBSD adds the port to the tree officially.  That would conclude adding official FreeBSD support for 3.4.0.
